src/dune_engine/scheduler.ml

```diff
@@ -782,17 +782,44 @@ end = struct
   ;;
 end
 
+module Trigger : sig
+  (** Conceptually, a [unit Fiber.Ivar.t] that may be filled more than once. *)
+  type t
+
+  val create : unit -> t
+  val trigger : t -> Fiber.fill list
+  val wait : t -> unit Fiber.t
+end = struct
+  type t =
+    { ivar : unit Fiber.Ivar.t
+    ; mutable filled : bool
+    }
+
+  let create () = { ivar = Fiber.Ivar.create (); filled = false }
+
+  let trigger t =
+    if not t.filled
```

Comment:
If you invert this `if`, it will be easier to read and the shorter clause will be on top.

Reply:
Oops, yup, I don't know why I did it this way.